File: php/features.php
<?php
declare(strict_types=1);

// LoremPicsum SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class LoremPicsumFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new LoremPicsumBaseFeature();
            case "test":
                return new LoremPicsumTestFeature();
            default:
                return new LoremPicsumBaseFeature();
        }
    }
}

File: py/features.py
# LoremPicsum SDK feature factory

from feature.base_feature import LoremPicsumBaseFeature
from feature.test_feature import LoremPicsumTestFeature


def _make_feature(name):
    features = {
        "base": lambda: LoremPicsumBaseFeature(),
        "test": lambda: LoremPicsumTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
